endpoints/setuid.go

```diff
@@ -148,6 +163,62 @@ func NewSetUIDEndpoint(cfg *config.Configuration, syncersByBidder map[string]use
 	})
 }
 
+// extractGDPRInfo looks for the GDPR consent string and GDPR signal in the GPP query params
+// first and the 'gdpr' and 'gdpr_consent' query params second.
+func extractGDPRInfo(query url.Values) (gdpr.RequestInfo, error) {
+	var gdprSignal gdpr.Signal = gdpr.SignalNo
+	var gdprConsent string = query.Get("gdpr_consent")
+	var signalSet bool = false
+
+	// Signal
+	gppSID, err := stringutil.StrToInt8Slice(query.Get("gpp_sid"))
+	if err != nil {
+		return gdpr.RequestInfo{}, fmt.Errorf("Error parsing gpp_sid %s", err.Error())
+	}
+	if len(gppSID) > 0 && gppPrivacy.IsSIDInList(gppSID, gppConstants.SectionTCFEU2) {
+		gdprSignal = gdpr.SignalYes
+		signalSet = true
+	}
+
+	if !signalSet {
+		gdprQuerySignal := query.Get("gdpr")
+		if gdprQuerySignal != "" && gdprQuerySignal != "0" && gdprQuerySignal != "1" {
+			return gdpr.RequestInfo{}, errors.New("the gdpr query param must be either 0 or 1. You gave " + gdprQuerySignal)
+		}
+
+		if gdprQuerySignal == "1" && gdprConsent == "" {
+			return gdpr.RequestInfo{}, errors.New("gdpr_consent is required when gdpr=1")
+		}
+		if i, err := strconv.Atoi(gdprQuerySignal); err == nil {
```

Comment:
If the GDPR signal was not present in the GPP query params, you're looking at the legacy GDPR query params. If `gdprQuerySignal` was not specified, we should be defaulting `gdprSignal` to `SignalAmbiguous` as you'll see in the code you deleted below. The `perms` object instantiated via `permsBuilder` in `preventSyncsGDPR` will properly handle the ambiguous signal.

Reply:
You're completely right. Made major corrections to this function in the next commit.